java/src/jmri/jmrit/dispatcher/AutoActiveTrain.java

```diff
@@ -898,11 +898,14 @@ private void setSpeedBySectionsAllocated() {
                     // .getSpeed(InstanceManager.getDefault(DispatcherFrame.class).getStoppingSpeedName());
                     _activeTrain.setStatus(ActiveTrain.RUNNING);
             }
-            for (Block block : _currentAllocatedSection.getSection().getBlockList()) {
-                float speed = -1.0f;
-                speed = getSpeedFromBlock(block);
-                if (speed > 0 && speed < newSpeed) {
-                    newSpeed = speed;
+            // If the train has no _currentAllocatedSection it is in a first block outside transit.
+            if (_currentAllocatedSection != null ) {
+                for (Block block : _currentAllocatedSection.getSection().getBlockList()) {
+                    float speed = -1.0f;
+                    speed = getSpeedFromBlock(block);
```

Comment:
The first value of speed, -1.0f, is never used. So it's better to combine these two lines:
`float speed = getSpeedFromBlock(block);`